src/main/java/appeng/api/stacks/AEItemKey.java

```diff
@@ -25,9 +25,11 @@ public final class AEItemKey extends AEKey {
     @Nullable
     private final CompoundTag tag;
     private final int hashCode;
+    private final Component displayName;
 
     private AEItemKey(Item item, @Nullable CompoundTag tag) {
         this.item = item;
+        this.displayName = toStack().getHoverName();
```

Comment:
Obviously this needs to be moved below `this.tag = tag`. :P

Reply:
I made some logical changes.

- First remove AEKey#getDisplayName implementation in AEFluidKey and AEItemKey, implement the method in AEKey. So the sub-class should super one displayName field.

- Second I have reserved no-arg constructor to avoid compatibility problems. The method that implemented this class before must implement the getDisplayName method.No need to worry about null return.

- Third, add Platform#getItemDisplayName to get item display name by itself and its tag. And change Platform#getFluidDisplayName, the argument is replaced by the Fluid instance

Reply:
Looks good, just one last thing to address.